dango/bitcoin/src/execute.rs

```diff
@@ -0,0 +1,678 @@
+
+    // TODO: remove this check?
+    ensure!(
+        recipient != cfg.vault,
+        "cannot withdraw to the vault address"
+    );
+
+    // If there is already a withdrawal to the same recipient, accumulate the amount.
+    OUTBOUND_QUEUE.may_update(ctx.storage, recipient, |outbound| -> StdResult<_> {
+        Ok(outbound.unwrap_or_default().checked_add(amount)?)
+    })?;
+
+    Ok(Response::new())
+}
+
+fn create_deposit_address(ctx: MutableCtx) -> anyhow::Result<Response> {
+    let (index, _) = ADDRESS_INDEX.increment(ctx.storage)?;
+
+    // Add a deposit address for the user if he hasn't already one.
+    ADDRESSES.may_update(
+        ctx.storage,
+        ctx.sender,
+        |maybe_address| match maybe_address {
+            Some(_) => bail!("you already have a deposit address"),
+            None => Ok(index),
+        },
+    )?;
+
+    Ok(Response::new())
+}
+
+#[cfg_attr(not(feature = "library"), grug::export)]
+pub fn cron_execute(ctx: SudoCtx) -> anyhow::Result<Response> {
+    let cfg = CONFIG.load(ctx.storage)?;
+
+    // Take the pending outbound transfers from the storage.
+    let outputs = OUTBOUND_QUEUE.drain(ctx.storage, None, None)?;
+
+    // If there's no pending outbound transfers, nothing to do.
+    if outputs.is_empty() {
+        return Ok(Response::new());
+    }
+
+    // Each transaction can have a maximum number of outputs.
+    let mut events = vec![];
+    let mut iter = outputs.into_iter();
+
+    // Keep creating transactions until there are no more outputs left.
+    while iter.len() > 0 {
+        let mut tx_output = BTreeMap::new();
+        let mut withdraw_amount = Uint128::ZERO;
+
+        for (k, v) in iter.by_ref() {
+            tx_output.insert(k, v);
+            withdraw_amount += v;
+
+            // Check if the maximum number of outputs is reached for this tx.
+            if tx_output.len() == cfg.max_output_per_tx {
+                break;
+            }
+        }
+
+        // Choose the best UTXOs for this transaction.
+        let transaction_builder = select_best_utxos(ctx.storage, cfg.clone(), tx_output.clone())?;
+
+        // If there's excess input, send the excess back to the vault.
+        let surplus_amount = transaction_builder.surplus_amount();
+        // TODO: should we check that the surplus amount is above the minimum deposit?
+        // If so, the fees will be higher and the tx could be faster.
+        if surplus_amount > Uint128::ZERO {
+            tx_output.insert(cfg.vault.clone(), surplus_amount);
+        }
+
+        // Delete the chosen UTXOs.
+        for ((hash, vout), (amount, _)) in &transaction_builder.inputs {
+            UTXOS.remove(ctx.storage, (*amount, *hash, *vout));
+        }
+
+        let (id, _) = OUTBOUND_ID.increment(ctx.storage)?;
+        let fee = transaction_builder.fee();
+        let transaction = Transaction {
+            inputs: transaction_builder.inputs,
+            outputs: tx_output,
+            fee,
+        };
```

Comment:
I know we have the withdrawal fee in gateway contract, but would it make sense to instead make the withdrawal fee based on the actual cost of the withdrawal transaction? Can we end up in a situation where the withdrawal fee paid is less the transaction fee because e.g. it requires many small (value, not size) UTXOS to be spent?